### PR TITLE
Turn on no_enhanced_typecheck by default

### PR DESCRIPTION
**Pull request by Chris-Hawblitzel**
Next step towards using unmodified rustc: by default, disable Verus type checking extensions to rustc, unless --deprecated-enhanced-typecheck is set.

### CODE REVIEW

File: source/rust_verify/example/invariants.rs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+// rust_verify/tests/example.rs deprecated-enhanced-typecheck
```
Comment (utaal) [inline source/rust_verify/example/invariants.rs:1]: This implies `expect-success` right?

Comment (Chris-Hawblitzel) [reply source/rust_verify/example/invariants.rs:1]: Yes, because `expect-success` is the default.

File: source/rust_verify/tests/examples.rs [hunk 2]
```diff
@@ -56,12 +56,19 @@ fn run_example_for_file(file_path: &str) {
...
+    }
 
     if let ["//", "rust_verify/tests/example.rs", command] = &first_line_elements[..] {
         match command.strip_suffix("\n").unwrap_or("unexpected") {
             "expect-success" => mode = Mode::ExpectSuccess,
             "expect-errors" => mode = Mode::ExpectErrors,
             "expect-failures" => mode = Mode::ExpectFailures,
+            "deprecated-enhanced-typecheck" => {
+                deprecated_enhanced_typecheck = "--deprecated-enhanced-typecheck";
+            }
```
Comment (tjhance) [inline source/rust_verify/tests/examples.rs:71]: could you clean this up? the other 3 cases each set the `mode` variable, while this one leaves it as the default value, set several lines above.

**Review by tjhance — APPROVED** [review]

File: source/rust_verify/tests/examples.rs (hunk 2 above)
Comment (utaal) [reply source/rust_verify/tests/examples.rs:71]: can you also please add a TODO/REVIEW to remove this option once we've removed `--deprecated-enhanced-typecheck`?

**Review by utaal — APPROVED** [review]

**Pull request merged**